testutil/ciao-down/workloads/clearcontainers-3.x.yaml

```diff
@@ -0,0 +1,249 @@
+ - {{template "ENV" .}} sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+ - {{endTaskCheck .}}
+
+ - {{beginTask . "Add Clear Containers OBS Repository "}}
+ - sudo sh -c "echo 'deb http://download.opensuse.org/repositories/home:/clearcontainers:/clear-containers-3/xUbuntu_16.04/ /' >> /etc/apt/sources.list.d/clear-containers.list"
+ - {{template "ENV" .}}curl -fsSL http://download.opensuse.org/repositories/home:/clearcontainers:/clear-containers-3/xUbuntu_16.04/Release.key | sudo apt-key add -
+ - {{endTaskCheck .}}
+
+ - {{beginTask . "Add CRI-O OS dependency repo"}}
+ - {{template "ENV" .}}sudo add-apt-repository "ppa:alexlarsson/flatpak"
+ - {{endTaskCheck .}}
+
+ - {{beginTask . "Add CRI-O repo"}}
+ - {{template "ENV" .}}sudo add-apt-repository "ppa:projectatomic/ppa"
+ - {{endTaskCheck .}}
+
+ - {{beginTask . "Add K8S repo"}}
+ - sudo sh -c "echo 'deb http://apt.kubernetes.io/ kubernetes-xenial-unstable main' >> /etc/apt/sources.list.d/kubernetes.list"
+ - {{template "ENV" .}}curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
+
+ - {{beginTask . "Retrieving updated list of packages"}}
+ - {{template "ENV" .}}sudo apt-get update
+ - {{endTaskCheck .}}
+
+ - {{beginTask . "Installing Clear Containers Runtime"}}
+ - {{template "ENV" .}}sudo apt-get install -y cc-runtime cc-proxy cc-shim
+ - {{endTaskCheck .}}
+
+ - {{beginTask . "Installing Docker"}}
+ - {{template "ENV" .}}sudo apt-get install -y --allow-downgrades --allow-unauthenticated docker-engine=1.12.1-0~xenial
+ - {{endTaskCheck .}}
+
+ - {{beginTask . "Start Clear Containers Runtime"}}
+ - sudo systemctl daemon-reload
+ - sudo systemctl restart docker
+ - sudo systemctl enable cc-proxy.socket
+ - sudo systemctl start cc-proxy.socket
+ - {{endTaskCheck .}}
+
+ - {{beginTask . "Installing GCC"}}
+ - {{template "ENV" .}}apt-get install gcc -y
+ - {{endTaskCheck .}}
+
+ - {{beginTask . "Installing Make"}}
+ - {{template "ENV" .}}apt-get install make -y
+ - {{endTaskCheck .}}
+
+ - {{beginTask . "Installing QEMU" }}
+ - {{template "ENV" .}}apt-get install qemu-system-x86 -y
+ - {{endTaskCheck .}}
+
+ - {{beginTask . "Installing xorriso"}}
+ - {{template "ENV" .}}apt-get install xorriso -y
+ - {{endTaskCheck .}}
+
+ - {{beginTask . "Installing Clear Containers development tools"}}
+ - {{template "ENV" .}}apt-get install build-essential python zlib1g-dev libcap-ng-dev libglib2.0-dev libpixman-1-dev libattr1-dev libcap-dev autoconf libtool libjson-glib-dev uuid-dev check bats libdevmapper-dev file apt-utils wget valgrind lcov libmnl-dev cppcheck libtap-formatter-html-perl -y
+ - {{endTaskCheck .}}
+
+ - {{beginTask . "Installing CRI-O dependencies"}}
+ - {{template "ENV" .}}apt-get install -y socat libgpgme11 libostree-1-1
+ - {{endTaskCheck .}}
+
+ - {{beginTask . "Installing K8S"}}
+ - {{template "ENV" .}}apt-get install -y kubelet=1.7.5-00 kubeadm=1.7.5-00 kubectl=1.7.5-00
```

Comment:
We had problems with installing things from the k8s repos in kubicle.  Sometimes we found that after adding the k8s repo to our sources  apt-get update or an installation of one of the components would fail.  Perhaps we were just unlucky and we were testing kubicle during a week when the k8s servers were having difficulties.  Nonetheless, we modified kubicle to retry apt commands related to k8s after this.  See here:

https://github.com/01org/ciao/blob/master/k8s/kubicle/cloudinit.go#L83

If you run into similar problems you could try a similar approach

Reply:
Good to know -- I haven't run into any issues yet, FWIW.  Noted.